packages/indexer-common/src/transactions.ts

```diff
@@ -0,0 +1,396 @@
+import {
+  BigNumber,
+  BigNumberish,
+  ContractReceipt,
+  ContractTransaction,
+  providers,
+  utils,
+  Wallet,
+} from 'ethers'
+import {
+  Address,
+  Eventual,
+  Logger,
+  mutable,
+  NetworkContracts,
+  timer,
+  toAddress,
+} from '@graphprotocol/common-ts'
+import delay from 'delay'
+import { IndexerError, indexerError, IndexerErrorCode } from './errors'
+import { TransactionConfig, TransactionType } from './types'
+import { NetworkSubgraph } from './network-subgraph'
+import gql from 'graphql-tag'
+
+export class TransactionManager {
```

Comment:
Is this where we expect the Action Queue implementation will hook in?

Reply:
Largely yes; I expect the action queue worker implementation (that picks up action items and executes them) will essentially hook in here.   Additionally there will be graphql resolvers for adding items to the action queue that the indexer-agent (or external tools) will call in place of it's current transaction execution.